src/tools/diagnostics.md

```diff
@@ -0,0 +1,2436 @@
+---
+title: Diagnostics
+description: Details for diagnostics produced by the Dart analyzer.
+---
+
+This page lists all diagnostic messages produced by the Dart analyzer,
+with details about what those messages mean
+and how you can fix your code.
+For more information about the analyzer, see
+[Customizing static analysis](/guides/language/analysis-options).
+
+
+## Glossary
+
+This page uses the following terms.
+
+### Element
+
+[PENDING: definition]
+
+### Potentially non-nullable
+
+A type is _potentially non-nullable_ if it's either explicitly non-nullable or
+if it's a type parameter.
+The latter case is included because
+the actual runtime type might be non-nullable.
+
+
+## Compile-time errors
```

Comment:
I seem to remember that we had decided not to group diagnostics by their severity. I had those headers in the doc because it made it easier for me to keep track of which diagnostics already had documentation, not because I thought they belonged in the output.

The strongest argument I can remember against grouping diagnostics by severity is that users can change the severities in the analysis options file, so the published severity won't necessary match what they see in the IDE and command line output.

Also, given the way I expect users will interact with this page, I don't believe that grouping the diagnostics (in any way) will be useful.

I am, of course, happy to discuss the question. But if you agree that we don't want to group by severity, then I need to know whether we want a second-level heading before the diagnostics (to align with "Glossary") or whether each diagnostic name will be a second level heading.

Reply:
I'm fine with not grouping by severity. They should probably be alphabetical. I'd keep them as third-level headings, so we'd have two second-level headings:

* Glossary
* Diagnostics

Reply:
Will do.